### PR TITLE
updated gift card selection default options for multiple giftcards

### PR DESCRIPTION
**Pull request by AnuragBhardwaj1**

### CODE REVIEW

File: app/controllers/spree/gift_cards_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module Spree
   class GiftCardsController < Spree::StoreController
 
+    before_filter :load_variant
     def new
```
Comment (tanmay3011) [inline app/controllers/spree/gift_cards_controller.rb]: @AnuragBhardwaj1 Add a line

File: app/controllers/spree/gift_cards_controller.rb [hunk 2]
```diff
@@ -45,5 +46,9 @@ def gift_card_params
       params.require(:gift_card).permit(:email, :name, :note, :variant_id)
     end
 
+    def load_variant
+      @current_variant = Spree::Product.friendly.find(params[:id]).try(:variants_including_master).try(:first) if (params[:id]).present?
```
Comment (tanmay3011) [inline app/controllers/spree/gift_cards_controller.rb]: @AnuragBhardwaj1 Move Spree::Product.friendly.find(params[:id]) to a separate before_action load_product

File: app/controllers/spree/products_controller_decorator.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@
   private
 
     def redirect_gift_card
-      redirect_to new_gift_card_path and return false if @product.try :is_gift_card?
+      redirect_to new_gift_card_path(id: params[:id]) and return false if @product.try :is_gift_card?
```
Comment (tanmay3011) [inline app/controllers/spree/products_controller_decorator.rb]: @AnuragBhardwaj1 Avoid sending params[:id] instead use the instance_variable loaded in before_action methods

**Comment by sawangupta92** [comment]
This PR is no longer required.

**Pull request closed without merging**